cli/internal/specs/v0/registry.go

```diff
@@ -1,56 +1,34 @@
 package specs
 
 import (
-	"bytes"
 	"encoding/json"
 	"fmt"
 )
 
-type Registry int
+type Registry string
 
 const (
-	RegistryGithub Registry = iota
-	RegistryLocal
-	RegistryGrpc
-	RegistryDocker
-	RegistryCloudQuery
+	RegistryGithub     = Registry("github")
+	RegistryLocal      = Registry("local")
+	RegistryGrpc       = Registry("grpc")
+	RegistryDocker     = Registry("docker")
+	RegistryCloudQuery = Registry("cloudquery")
 )
 
-func (r Registry) String() string {
-	return [...]string{"github", "local", "grpc", "docker", "cloudquery"}[r]
-}
-
-func (r Registry) MarshalJSON() ([]byte, error) {
-	buffer := bytes.NewBufferString(`"`)
-	buffer.WriteString(r.String())
-	buffer.WriteString(`"`)
-	return buffer.Bytes(), nil
-}
-
 func (r *Registry) UnmarshalJSON(data []byte) (err error) {
 	var registry string
 	if err := json.Unmarshal(data, &registry); err != nil {
 		return err
 	}
-	if *r, err = RegistryFromString(registry); err != nil {
-		return err
-	}
-	return nil
+	*r = Registry(registry)
+	return r.IsValid()
```

Comment:
This make sure if `registry` is specified it needs to be a valid value. It still stays zero value (empty string) if it's not specified.